### PR TITLE
chore, fix, feat(egen): follows template + fix docker run + adds clean up

### PR DESCRIPTION
**Pull request by krishr2d2**
<br>
The notebook's optional docker run (local) step fails when executed in the Vertex AI Workbench. Removing sudo from the command has fixed the issue. Along with the command update, this PR includes:

1. Adds Colab Enterprise link 
2. Rewords some sentences to match the guidelines
3. Rewords some headings as per the content and fixes their styles as per the hierarchy
4. Removes IS_TESTING from the cleaning up section
5. Adds clean up step for locally created files
<br><br><br>

**REQUIRED:** Fill out the below checklists or remove if irrelevant
1. If you are opening a PR for `Official Notebooks` under the [notebooks/official](https://github.com/GoogleCloudPlatform/vertex-ai-samples/tree/main/notebooks/official) folder, follow this mandatory checklist:
- [x] Use the [notebook template](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/notebook_template.ipynb) as a starting point.
- [x] Follow the style and grammar rules outlined in the above notebook template.
- [x] Verify the notebook runs successfully in Colab since the automated tests cannot guarantee this even when it passes.
- [x] Passes all the required automated checks. You can locally test for formatting and linting with these [instructions](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/CONTRIBUTING.md#code-quality-checks).
- [ ] You have consulted with a tech writer to see if tech writer review is necessary. If so, the notebook has been reviewed by a tech writer, and they have approved it.
- [ ] This notebook has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/official/CODEOWNERS) file under the `Official Notebooks` section, pointing to the author or the author's team.
- [x] The Jupyter notebook cleans up any artifacts it has created (datasets, ML models, endpoints, etc) so as not to eat up unnecessary resources.

<br>

### CODE REVIEW

File: notebooks/official/custom/SDK_Custom_Container_Prediction.ipynb [hunk 1]
```diff
@@ -855,7 +843,7 @@
         "id": "2d242773d707"
       },
       "source": [
-        "#### Enable Artifact Registry API\n",
+        "#### Enable artifact registry API\n",
```
Comment (kittyabs) [inline notebooks/official/custom/SDK_Custom_Container_Prediction.ipynb:846]: Why was the case changed here?

File: notebooks/official/custom/SDK_Custom_Container_Prediction.ipynb [hunk 2]
```diff
@@ -882,12 +870,12 @@
         "id": "23f3ba8346f4"
       },
       "source": [
-        "### Create a private Docker repository\n",
+        "#### Create a docker repository in artifact registry\n",
```
Comment (kittyabs) [inline notebooks/official/custom/SDK_Custom_Container_Prediction.ipynb:873]: again, should be "Artifact Registry"

**Review by kittyabs — COMMENTED** [review]
several suggested edits

**Review by katiemn — APPROVED** [review]

**Pull request merged**